### ISSUE TITLE
Multiple different auth headers?

### ISSUE THREAD

**Issue by LasseGravesen** [tiangolo/fastapi#741]
**Description**

Is it possible to set multiple different, optional auth headers using the security functionality?

**Additional context**
This bit of code:
```
X_APIKEY=APIKeyHeader(name="X-APIKey", auto_error=False)
X_USER_APIKEY = APIKeyHeader(name="X-User-APIKey", auto_error=False)

@router.get("/ping")
def ping(
    request: Request,
    x_api_key: str =  Depends(X_API_KEY),
    x_user_apikey: str = Depends(X_USER_APIKEY),
):
```

Essentially just overwrites the previous security header. Is there a good way to do this?

**Comment by tiangolo** [comment]
Yes @LasseGravesen , it's just not well documented yet. But just what you did, more or less.

You would put that in a dependency. And in the dependency you would check if one or the other auth method provided the credentials. And if none of them did, raise an `HTTPException`.

**Comment by ericman93** [comment]
I was trying to have it in a dependency, it worked great but in the docs when clicking on Authorize I see only 1 header (instead of the possible 3)

**Comment by Simon089** [comment]
> I was trying to have it in a dependency, it worked great but in the docs when clicking on Authorize I see only 1 header (instead of the possible 3)

If you provide a (different) `scheme_name` to `APIKeyHeader` both will show up in the authorize section. 

The first two lines from the initial question above could therefore be:
```
X_APIKEY=APIKeyHeader(name="X-APIKey", auto_error=False, scheme_name="API Key")
X_USER_APIKEY = APIKeyHeader(name="X-User-APIKey", auto_error=False, scheme_name="User API Key")
```
